src/_guides/language/evolution.md

```diff
@@ -313,74 +375,75 @@ multiple versions of the Dart language.
 The compiler determines what version the code is targeting,
 and it interprets the code according to that version.
 
-Language versioning is important on the rare occasions
-when Dart introduces an incompatible feature like [null safety][].
-Code that used to compile cleanly before null safety
-(but perhaps crash at runtime)
-might no longer compile once null safety is enabled.
-Because migrating your apps and packages—and all the packages they depend on—to
-null safety might take a while,
-Dart uses language versioning to support
-using non-null-safe code alongside null-safe code.
-For an example of how an app or package can migrate to
-a new language version
-with an incompatible feature (null safety, for example), see 
+Language versioning becomes important on the rare occasions when Dart
+introduces an incompatible feature like [null safety][].
+When Dart introduces a breaking change, code that
+did compile might no longer compile.
+Language versioning allows you to set each library's language version
+to maintain compatibility.
+
+In the case of null safety, Dart SDKs 2.12 through 2.19 allowed you
+to _choose_ to update your code to use null safety.
+Dart uses language versioning to permit non-null-safe code to run
+alongside null-safe code.
+This decision enabled migration from non-null-safe to null-safe code.
+To review an example of how an app or package can migrate to a new
+language version with an incompatible feature, check out
 [Migrating to null safety](/null-safety/migration-guide).
 
-Each package has a default language version,
-equal to the `<major>.<minor>` part of the
-**lower SDK constraint** in the pubspec.
-For example, the following entry in a `pubspec.yaml` file
-indicates that this package uses the Dart 2.7 language version.
+Each package has a default language version
+equal to the **lower bound of the SDK constraint** in the `pubspec.yaml` file.
+
+**For example:** The following entry in a `pubspec.yaml` file
+indicates that this package uses the Dart 2.18 language version or later.
 
 ```yaml
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.18.0 <3.0.0'
 ```
 
-
 ### Language version numbers
 
-Dart language versions are identified by a major and minor number
-that match the first two components of the Dart SDK.
-For example, the latest language version supported by
-the 2.7.3 Dart SDK is Dart 2.7.
-Each Dart SDK supports all of the language versions covered
-by its major version number.
-That means that the 2.7.3 Dart SDK supports language
-versions 2.7, 2.6, 2.5, and so on, down to 2.0.
+Dart formats its language versions as two numbers separated with a period.
+It reads as a major version number and a minor version number.
+Though this appears to follow semantic versioning, it does not.
```

Comment:
Consider dropping this sentence, it doesn't add anything to the meaning and it just seems a bit scary.

```suggestion
```

Reply:
**comment:** I'll drop it, but I do think it makes any confusion unambiguous.